### PR TITLE
Add errors and timeouts to mock server

### PR DESCRIPTION
**Pull request by diachenko-mischa**
Add emulation of errors and timeouts for mock server. This behavior is
controlled by 'debug' header in http requests. Network wrapper for
performance test implements interface to set this header.

Relates-To: OLPEDGE-1115

### CODE REVIEW

File: tests/utils/olp_server/server.js [hunk 1]
```diff
@@ -29,11 +29,80 @@ const blob_service_handler = require('./blob_service.js')
...
 
-const handlers = {}
+function timeoutDecorator(processor) {
+  return function(response, pathname, request, handler) {
+    milliseconds = Math.floor(Math.random() * 250);
+    console.log('Waiting for ' + milliseconds + 'ms');
+
+    var waitTill = new Date(new Date().getTime() + milliseconds);
+    while (waitTill > new Date()) {
+    }
```
Comment (mykhailo-kuchma) [inline tests/utils/olp_server/server.js]: can we use sleep here?

File: tests/performance/NetworkWrapper.h [hunk 2]
```diff
@@ -43,17 +46,43 @@ class Http2HttpNetworkWrapper : public Network {
...
 
   void Cancel(RequestId id) override { network_->Cancel(id); }
 
-  Http2HttpNetworkWrapper()
-      : network_{std::move(olp::http::CreateDefaultNetwork(32))} {}
+  /*
+   * Adds special header, which signal mock server to generate timeouts and
+   * stalls when serving requests.
+   */
+  void SetTimeouts(bool with_timeouts) { with_timeouts_ = with_timeouts; }
```
Comment (borzun) [inline tests/performance/NetworkWrapper.h]:
```suggestion
  void EnableTimeouts(bool with_timeouts) { with_timeouts_ = with_timeouts; }
```

File: tests/performance/NetworkWrapper.h [hunk 3]
```diff
@@ -43,17 +46,43 @@ class Http2HttpNetworkWrapper : public Network {
...
+
+  /*
+   * Adds special header, which signal mock server to generate errors when
+   * serving requests. Error rate is 10%.
+   */
+  void SetErrors(bool with_errors) { with_errors_ = with_errors; }
```
Comment (borzun) [inline tests/performance/NetworkWrapper.h]:
```suggestion
  void EnableErrors(bool with_errors) { with_errors_ = with_errors; }
```

File: tests/performance/NetworkWrapper.h [hunk 4]
```diff
@@ -43,17 +46,43 @@ class Http2HttpNetworkWrapper : public Network {
...
 
  private:
+  std::string GetDebugHeader() {
```
Comment (borzun) [inline tests/performance/NetworkWrapper.h]:
```suggestion
  std::string GetDebugHeader() const {
```

File: tests/utils/olp_server/server.js [hunk 5]
```diff
@@ -29,11 +29,80 @@ const blob_service_handler = require('./blob_service.js')
...
+
+
+function handleRequest(response, pathname, request, handler) {
+  const result = handler(pathname, request)
+  response.writeHead(result.status, result.headers)
+  response.end(result.text)
+}
+
+function
+getError() {
```
Comment (borzun) [inline tests/utils/olp_server/server.js]:
```suggestion
generateErrorAtRandom() {
```
Or `getErrorAtRandom`. IMHO, `getError` is too basic name

File: tests/performance/NetworkWrapper.h [hunk 6]
```diff
@@ -43,17 +46,43 @@ class Http2HttpNetworkWrapper : public Network {
...
+    if (with_errors_) {
+      debug_header_value += "errors,";
+    }
+
+    if (with_timeouts_) {
+      debug_header_value += "timeouts,";
+    }
+
+    return debug_header_value;
+  };
```
[the remark this answers is not in the document]
Comment (borzun) [reply tests/performance/NetworkWrapper.h]: What do you mean by that?

File: tests/utils/olp_server/server.js (hunk 1 above)
Comment (diachenko-mischa) [reply tests/utils/olp_server/server.js]: sleep doesn't work

File: tests/performance/NetworkWrapper.h (hunk 6 above)
[the remark this answers is not in the document]
Comment (diachenko-mischa) [reply tests/performance/NetworkWrapper.h]: Do you mean compute header on WithErrors/WithTimeout changes and use computed value here?

File: tests/utils/olp_server/server.js (hunk 1 above)
Comment (diachenko-mischa) [reply tests/utils/olp_server/server.js]: Fixed available sleep function

**Pull request merged**